google_photos_takeout_helper/__main__.py

```diff
@@ -276,38 +301,52 @@ def find_json_for_file(file: Path):
             raise FileNotFoundError(f"Couldn't find json for file: {file}")
 
     # Returns date in 2019:01:01 23:59:59 format
-    def get_date_from_folder_name(dir: Path):
-        dir = dir.name
-        dir = dir[:10].replace('-', ':').replace(' ', ':') + ' 12:00:00'
-
-        # Sometimes google exports folders without the -, like 2009 08 30...
-        # So the end result would be 2009 08 30 12:00:00, which does not match the format.
-        # Therefore, we also replace the spaces with ':'
-
-        # Reformat it to check if it matcher, and quit if doesn't match - it's probably a date folder
+    def get_date_from_folder_meta(dir: Path):
         try:
-            return _datetime.strptime(dir, '%Y:%m:%d %H:%M:%S').strftime('%Y:%m:%d %H:%M:%S')
-        except ValueError as e:
-            print()
-            print(e)
-            print()
-            print('==========!!!==========')
-            print(f"Wrong folder name: {dir}")
-            print("You probably forgot to remove 'album folders' from your takeout folder")
-            print("Please do that - see README.md or --help for why")
-            print("https://github.com/TheLastGimbus/GooglePhotosTakeoutHelper#why-do-you-need-to-cut-out-albums")
-            print()
-            print('Once you do this, just run it again :)')
-            print('==========!!!==========')
-            exit(-1)
+            file = find_album_meta_json_file(dir)
+            if file:
+                try:
+                    with open(str(file), 'r') as f:
+                        dict = _json.load(f)
+                        if "date" in dict["albumData"]:
+                            if "timestamp" in dict["albumData"]["date"]:
+                                return _datetime.fromtimestamp(int(dict["albumData"]["date"]["timestamp"])).strftime(
+                                    '%Y:%m:%d %H:%M:%S')
+                except:
+                    pass
+        except:
+            pass
+
+        print("Couldn't pull datetime from album meta")
+        return None
+
+    def find_album_meta_json_file(dir: Path):
+        if str(dir) in meta_file_memo:
+            return meta_file_memo[str(dir)]
+
+        for file in dir.rglob("*.json"):
+            try:
+                with open(str(file), 'r') as f:
+                    dict = _json.load(f)
+                    if "albumData" in dict:
+                        meta_file_memo[str(dir)] = file
+                        return file
+            except Exception as e:
+                print(e)
+                raise FileNotFoundError(f"find_album_meta_json_file - Couldn't find json for file: {file}")
```

Comment:
This just iterates over all `.json`s, and returns one with `"albumData"`

Then why abondon everything when one of them is corrupted/something

```suggestion
                print(f"find_album_meta_json_file - Error opening file: {file}")
```
Now, we don't need to wrap this in `try` at all - just check if it `is None`